src/server/api/api.ts

```diff
@@ -237,24 +260,21 @@ apiRouter.get("/get-simulator-robot-state", (_, res) => {
     });
 });
 
+export interface PuzzleComponents {
+    fen: string;
+    moves: Move[];
+    rating: number;
+}
+
 /**
- * Returns a list of available puzzles to play.
+ * Returns a list of available puzzles to play from puzzles.json.
  */
 apiRouter.get("/get-puzzles", (_, res) => {
-    return res.send({
-        puzzles: [
-            {
-                name: "Puzzle 1",
-                id: "puzzleId1",
-                rating: "1200",
-            },
-            {
-                name: "Puzzle 2",
-                id: "puzzleId2",
-                rating: "1400",
-            },
-        ],
-    });
+    const puzzles: Map<string, PuzzleComponents> = JSON.parse(
```

Comment:
this type should use a [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) since we aren't creating a JS [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)

there's a couple other places that a `Map` type is used on puzzles to represent what is actually just a regular JS object, so those should also be changed